OpenRA.Game/Server/Server.cs

```diff
@@ -1011,12 +1177,62 @@ public void StartGame()
 			// TODO: Enable for multiplayer (non-dedicated servers only) once the lobby UI has been created
 			LobbyInfo.GlobalSettings.GameSavesEnabled = Type != ServerType.Dedicated && LobbyInfo.NonBotClients.Count() == 1;
 
+			if (recorder != null)
+			{
+				gameInfo = new GameInformation
+				{
+					Mod = Game.ModData.Manifest.Id,
+					Version = Game.ModData.Manifest.Metadata.Version,
+					MapUid = Map.Uid,
+					MapTitle = Map.Title,
+					StartTimeUtc = DateTime.UtcNow
+				};
+
+				recorder.Metadata = new ReplayMetadata(gameInfo);
+				foreach (var conn in Conns)
+				{
+					var client = GetClient(conn);
+					if (client.IsObserver)
+						continue;
+
+					// TODO: Populate metadata with real values for:
+					//   - FactionName
+					//   - FactionId (requires game knowledge)
+					//   - SpawnPoint
+					//   - IsRandomFaction
+					//   - IsRandomSpawnPoint
+					var player = new GameInformation.Player
+					{
+						ClientIndex = client.Index,
+						Name = client.Name,
+						IsHuman = client.Bot == null,
+						IsBot = client.Bot != null,
+						FactionName = "Random",
```

Comment:
You can then use `clientFaction.Name` and `clientFaction.InternalName` here, which makes for a better placeholder for random factions and the actual correct values for non-random ones.

Reply:
Right, adjusted, thanks again.